code/modules/surgery/organs/augments_internal.dm

```diff
@@ -117,19 +116,16 @@
 	. = ..()
 	RegisterSignal(owner, signalCache, .proc/on_signal)
 
-/obj/item/organ/cyberimp/brain/anti_stun/proc/on_signal()
-	if(broken_cyber_organ || working)
-		return
-	working = TRUE
-	if(owner.AmountStun() > stun_cap_amount)
-		owner.SetStun(stun_cap_amount)
-	if(owner.AmountKnockdown() > stun_cap_amount)
-		owner.SetKnockdown(stun_cap_amount)
-	if(owner.AmountImmobilized() > stun_cap_amount)
-		owner.SetImmobilized(stun_cap_amount)
-	if(owner.AmountParalyzed() > stun_cap_amount)
-		owner.SetParalyzed(stun_cap_amount)
-	working = FALSE
+/obj/item/organ/cyberimp/brain/anti_stun/proc/on_signal(datum/source, amount)
+	if(!broken_cyber_organ && amount > 0)
```

Comment:
what does the amount var do and does it change?

Reply:
Stun reductions (for example from chems) send us a negative amount which would reset the timer if this check weren't here.

Basically, nicotine breaks the implant without this.

Reply:
lovely